base/test.jl

```diff
@@ -22,6 +22,30 @@ export GenericString
 
 #-----------------------------------------------------------------------
 
+# Backtrace utility functions
+function ip_matches_name(ip, dir::String, file::String)
+    for fr in StackTraces.lookup(ip)
+        if fr === StackTraces.UNKNOWN || fr.from_c
+            return false
+        end
+        path = string(fr.file)
+        dirname(path) == dir && basename(path) == file && return true
+    end
+    return false
+end
+
+function scrub_backtrace(bt)
+    do_test_ind = findfirst(addr->Base.REPL.ip_matches_func(addr, :do_test), bt)
```

Comment:
i doubt that's guaranteed to always be true, `-g0` or things getting debug info stripped or using different versions of some libraries